### PR TITLE
Add nightly coverity scan

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR adds nightly coverity scan job to the FreeRTOS-Kernel which is scheduled to be run at 1:15 AM UTC daily.
Builds the minimal [cmake_example](https://github.com/FreeRTOS/FreeRTOS-Kernel/tree/main/examples/cmake_example) project using cov_build and uploads to [coverity scan](https://scan.coverity.com/projects/freertos-kernel) for analysis.

Test Steps
-----------
Tested on fork:

- [Example run](https://github.com/tony-josi-aws/FreeRTOS-Kernel/actions/runs/6637851834/job/18032992133)

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- ~[ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.~ NA

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by amazonKamath — DISMISSED** [review]

File: .github/workflows/coverity_scan.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+      - name: Coverity Build & Upload for Scan
+        shell: bash
+        env:
+          COVERITY_TOKEN: ${{ secrets.COVERITY_SCAN_TOKEN }}
+        run: |
+          export PATH="$PATH:${{env.cov_scan_path}}"
+          mkdir build
+          cd build
+          cmake ../examples/cmake_example/
```
Comment (chinglee-iot) [inline .github/workflows/coverity_scan.yml:38]:
```suggestion
          cmake -S ./examples/cmake_example/ -B build
          cd build
```

nit : we may consider to use `-S` to specify <path-to-source> and `-B` to specify the <path-to-build>.

**Review by chinglee-iot — APPROVED** [review]
Currently, this action still use personal account. It would be great if we can use non-personal account in the future.

**Review by chinglee-iot — APPROVED** [review]

File: .github/workflows/coverity_scan.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        env:
+          COVERITY_TOKEN: ${{ secrets.COVERITY_SCAN_TOKEN }}
+        run: |
+          export PATH="$PATH:${{env.cov_scan_path}}"
+          cmake -S ./examples/cmake_example/ -B build
+          cd build
+          cov-build --dir cov-int make -j
+          tar czvf gcc_freertos_kerenl_sample_build.tgz cov-int
+          COV_SCAN_UPLOAD_STATUS=$(curl --form token=${COVERITY_TOKEN} \
+            --form email=tonyjosi@amazon.com \
```
Comment (aggarg) [inline .github/workflows/coverity_scan.yml:41]: Approving assuming that this will be updated once the check is finalized.

**Review by aggarg — APPROVED** [review]

File: .github/workflows/coverity_scan.yml (hunk 2 above)
Comment (tony-josi-aws) [reply .github/workflows/coverity_scan.yml:41]: Yes, we can update it to a team email address.

**Pull request merged**